src/snap_management.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Learn more at: https://juju.is/docs/sdk
+
+"""Snap Installation Module.
+
+Modified from https://github.com/canonical/k8s-operator/blob/main/charms/worker/k8s/src/snap.py
+"""
+
+
+import logging
+import platform
+
+import charms.operator_libs_linux.v2.snap as snap_lib
+
+# Log messages can be retrieved using juju debug-log
+log = logging.getLogger(__name__)
+
+
+# Map of the grafana-agent snap revision to install for given architectures and strict mode.
+_grafana_agent_snap_name = "grafana-agent"
+_grafana_agent_snaps = {
+    # (confinement, arch): revision
+    ("strict", "amd64"): 16,
+    ("strict", "arm64"): 23,
+}
+
+
+class SnapSpecError(Exception):
+    """Custom exception type for errors related to the snap spec."""
+
+    pass
+
+
+def install_ga_snap(classic: bool = False):
```

Comment:
Just to be safe, should we change the "public api" to:
- `install_strict_snap()`
- `install_classic_snap()`
?
I.e. seems unsafe to have a default for "classic".

Reply:
Would you be happy if we just removed the default?  eg: `def install_ga_snap(classic: bool):`

I'm not sure if implementing separate install functions changes much.  It'll push the same logic into the install handler:
```python
    def _install(self) -> None:
        """Install/refresh the Grafana Agent snap."""
        self.unit.status = MaintenanceStatus("Installing grafana-agent snap")
        try:
            if classic:
              install_classic_snap()
            else:
              install_strict_snap()
        except (snap.SnapError, SnapSpecError) as e:
            raise GrafanaAgentInstallError("Failed to install grafana-agent.") from e
```

I'm not sure that feels any safer, just less convenient